### ISSUE TITLE
Session timer is not updated for non-refresh invite/response

### ISSUE THREAD

**Issue by danieludy** [drachtio/drachtio-server#197]
With drachtio as UAC and Session-Expires header in the initial 200 OK set to "1800;refresher=uas" drachtio behaves as expected and waits for UAS to refresh the session and will tear it down after 1800 seconds if nothing is received.

The issue is that if drachtio sends a reINVITE (not just a refresh but an actual modify) the UAS will include a new "Session-Expires: 1800;refresher=uas" and will restart the timer on the UAS but not within Drachtio. So ultimately Drachtio will tear down the call expecting to receive a refresh when one will not be sent...

Basically the same work needs to happen inside SipDialogController::processResponseInsideDialog that happens in SipDialogController::processResponseToRefreshingReinvite

**Comment by davehorton** [comment]
I've checked in a fix on the bugfix/issue-197 branch.  Any chance you can test it?

**Comment by danieludy** [comment]
While trying to fix it myself I hit an issue here as well. Just calling dlg->setSessionTimer is not enough as the previous timer still exists. I had to call dlg->cancelSessionTimer() before setting the new timer. I believe a better solution would be to just add that to the setSessionTimer itself to cancel any timer if it exists and then set the new one.

**Comment by davehorton** [comment]
good point.  I made that change, please test if you can
